pkg/resource/deploy/deployment_executor.go

```diff
@@ -399,6 +399,7 @@ func (ex *deploymentExecutor) performDeletes(
 	// is conservative, but correct.
 	erroredDeps := mapset.NewSet[*resource.State]()
 	seenErrors := mapset.NewSet[Step]()
+	ex.stepExec.ClearErroredSteps()
```

Comment:
Can't we just _not_ look at erroredSteps on line 404/5 below?

Reply:
ErroredSteps always contains the full set, and we need to look at them to know which dependencies we can't destroy.

Reply:
I'm not fully following... Is it because new errors from deletes could show up in ErroredSteps?

Reply:
Both errors from deletes and creates/updates can show up in the `ErroredSteps`. What we really need to deal with here is errors from failed deletes, as we need to not process their dependencies.  However errors for failed creates could still be in `ErroredSteps`, but would not be in the dependency graph.

I did change the code a little bit now though to just check if the errored resource is in the dependency graph, which avoids us having to clear out the errored steps, and feels a little better to me.